substrate-node/pallets/pallet-smart-contract/src/migrations/v8.rs

```diff
@@ -42,22 +42,22 @@ impl<T: Config> OnRuntimeUpgrade for FixTwinLockedBalances<T> {
     }
 }
 
-pub fn migrate_to_version_7<T: Config>() -> frame_support::weights::Weight {
+pub fn migrate_to_version_8<T: Config>() -> frame_support::weights::Weight {
     debug!(
         " >>> Starting contract pallet migration, pallet version: {:?}",
         PalletVersion::<T>::get()
     );
 
-    let mut read_writes = 0;
+    let mut reads = 0;
+    let mut writes = 0;
 
     let mut twin_contract_locked_balances: BTreeMap<u32, BalanceOf<T>> = BTreeMap::new();
-
+    // Fetch all locked balances based on the contract locks in storage and accumulate them by twin id
     for (ctr_id, l) in ContractLock::<T>::iter() {
         let ctr = Contracts::<T>::get(ctr_id);
-        read_writes += 1;
+        reads += 1;
```

Comment:
The code in the match also writes, no?

Reply:
It writes to a hashmap in memory, the weight counting is only for accessing the DB